pycontrails/core/vector.py

```diff
@@ -614,6 +629,7 @@ def __add__(self, other: Self | None) -> Self:
 
         return type(self).sum((self, other))
 
+    @final
```

Comment:
I think the final decorator tells tools like mypy that this method shouldn't be overridden by subclasses. But perhaps my usage isn't quite right. My thought was that we don't want to define something like `Flight() + Flight()`, because `Flight` instances have certain guarantees around the set of waypoints defining a single flight. I did add a specific type check in the method, so the final decorator does seem like overkill.

Reply:
That's my understanding of the semantics of `@final`, too. So is the point of `@final` plus the runtime check on the type of `cls` meant to
1. make sure `Flight` can't use the `sum` method it inherits from `VectorDataset`
2. make sure `Flight` will never get access to a `sum` method inherited from a subclass of `VectorDataset` that doesn't have a similar runtime check?

To me this is starting to smell like `Flight` shouldn't really be a subclass of `VectorDataset`... though it's probably not worth worrying about unless similar problems start to appear in other inherited methods.